appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/jmx/MetricsMetadataHelper.java

```diff
@@ -221,13 +221,70 @@ public List<MetricsMetadata> resolveDynamicMetadata(List<MetricsMetadata> metada
                 } finally {
                     removedMetadataList.add(metadata);
                 }
+            } else if (metadata.getService() != null
+                    && (metadata.getService().contains(SPECIFIER)
+                    || metadata.getService().contains(ATTRIBUTE)
+                    || metadata.getService().contains(SUB_ATTRIBUTE)
+                    || metadata.getService().contains(INSTANCE))) {
+                ServiceExpression expression = new ServiceExpression(
+                        metadata.getService().replace(SPECIFIER, "*")
+                );
+                if (expression.getAttributeName() != null
+                        && (expression.getAttributeName().equals("*") || expression.getAttributeName().equals(ATTRIBUTE))) {
+                    removedMetadataList.add(metadata);
+                    resolvedMetadataList.addAll(
+                            loadAttribute(expression, metadata)
+                    );
+                } else if (expression.getSubAttributeName() != null
+                        && (expression.getSubAttributeName().equals("*") || expression.getSubAttributeName().equals(SUB_ATTRIBUTE))) {
+                    removedMetadataList.add(metadata);
+                    resolvedMetadataList.addAll(
+                            loadSubAttribute(expression, metadata, expression.getAttributeName())
+                    );
+                }
             }
         }
 
         metadataList.removeAll(removedMetadataList);
         metadataList.addAll(resolvedMetadataList);
         return unresolvedMetadataList;
     }
+    
+    private List<MetricsMetadata> loadAttribute(
+            ServiceExpression expression,
+            MetricsMetadata metadata) {
+
+        List<MetricsMetadata> metadataList = new ArrayList<>();
+        String instanceName = serverEnv.getInstanceName();
+        HealthCheckStatsProvider healthCheck = habitat.getService(HealthCheckStatsProvider.class, expression.getServiceId());
```

Comment:
What about null handling here? Probably logging and skipping is the best approach.